lib/typhoeus/response/status.rb

```diff
@@ -81,6 +92,16 @@ def first_header_line
           end
         end
       end
+
+      # :nodoc:
+      def has_good_response_code?
+        response_code >= 200 && response_code < 300
```

Comment:
I don't think 3xx is an error either. For me the question if 3xx is an error or not highlights the issues I am seeing with that method. you pretty much have to know exactly what it is doing in order to use it. I don't always think a 404 is an error either for example. In other use cases it is. `sucess?` is easy to define since 2xx are the successful codes in http, but failure not so much. 

If we would decide on a definition of failure here and stick with it, I wouldn't mind it on the other hand. One can always check the codes manually.